Assets/Scripts/AStar/NodeMap.cs

```diff
@@ -0,0 +1,410 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Xml.Serialization;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+
+
+namespace AStar
+{
+    [Serializable]
+    public class NodeMap
+    {
+        public Node[,] nodes;
+        public int xNodeCount, yNodeCount;
+        public float xOrigin, yOrigin, zOrigin; //Origin starts at bottom left of map
```

Comment:
This could probably be a `Vector3`

Reply:
Vectors were intentionally avoided due to the way the NodeMap was saved. Long story short, all nested objects of NodeMap must have [Serializable] tag, which Unity Vectors does not have. The only way around this tag requirement to my knowledge is to simply not use it.